src/items/generics.md

```diff
@@ -83,6 +83,56 @@ where
 }
 ```
 
+## Free variables and concrete items and types
+
+The *free variables* of an item or type are the type and lifetime variables
+that refer to a generically declared parameter. A *concrete item* or *concrete
+type* is one that does not have any free variables. See [Wikipeida][concrete
+wikipedia] for more.
+
+For example, consider this trait:
+
+```rust
+# struct SomeStruct;
+#
+trait Example<A> {
+    fn foo(a: A);
+
+    fn bar<'b>(a: A, b: &'b SomeStruct);
+
+    fn baz();
+
+    fn quux(a: i32, b: Option<i32>);
+}
+```
+
+In it, the trait `Example` has a free variable `A` because it declares it
+itself. Furthermore, `foo` has `A` as a free variable because it uses it as the
+type of its first argument. `baz` has both `A` and `B` as free type variables,
+getting `A` from the trait's generic parameters and defining `B` as a generic
+parmaeter itself. Both `baz` and `quux` have no free type variables and are
+thusly concrete.
+
+For another example, the function `for<'a> fn foo(a: &'a i32>` is concrete.
+
+For another example, the following table shows the free type variables of
+various types. Types without free parmaeters show "concrete" instead of "none".
+Assume `A`, `E`, and `'a` are defined as generic parameters.
+
+| Type | Free Variables |
+| `i32` | concrete | 
+| `&'a i32` | `'a` |
+| `&'static i32` | concrete |
```

Comment:
Would be good to have `&i32` and say that it has an (anonymous) free (inference) variable.

Reply:
How's that?

Reply:
Well `&i32` really means `&'_ i32` and `_` is a placeholder for something that may be substituted or unified with something else, possibly concrete.